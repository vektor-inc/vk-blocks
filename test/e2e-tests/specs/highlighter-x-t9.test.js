/**
 * WordPress dependencies
 */
 import {
	pressKeyWithModifier,
	createNewPost,
	clickBlockAppender,
	clickBlockToolbarButton,
	getEditedPostContent,
	activateTheme,
	installTheme,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
 import { changeSiteLang } from '../helper/changeSiteLang';

describe( 'x-t9 Highlighter', () => {
  let oldLanguage;
	beforeAll( async () => {
		await installTheme( 'x-t9' );
		await activateTheme( 'x-t9' );
	} );
	beforeEach( async () => {
		oldLanguage = await changeSiteLang( 'en' );
		await createNewPost();
	} );
	afterEach( async () => {
		await changeSiteLang( oldLanguage );
	});
	afterAll( async () => {
		await activateTheme( 'twentytwentytwo' );
	} );

	it( 'x-t9 Highlighter', async () => {
		/**
		 * 新しく投稿を作る
		 */
		await createNewPost();

		await clickBlockAppender();

		// Add text and select to color.
		await page.keyboard.type( '1' );
		await pressKeyWithModifier( 'primary', 'a' );
		await clickBlockToolbarButton( 'More' );

		const button = await page.waitForXPath(
			`//button[text()='Highlighter']`
		);
		// Clicks may fail if the button is out of view. Assure it is before click.
		await button.evaluate( ( element ) => element.scrollIntoView() );
		await button.click();

		const option = await page.waitForSelector(
			'[aria-label="Color: Text Secondary"]'
		);

		await option.click();

		expect( await getEditedPostContent() ).toMatchSnapshot();

	} )

});
