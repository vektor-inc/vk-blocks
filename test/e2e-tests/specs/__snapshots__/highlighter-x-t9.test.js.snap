// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-t9 Highlighter x-t9 Highlighter 1`] = `
"<!-- wp:paragraph -->
<p><span data-color=\\"rgba(0,0,0,0.5)\\" style=\\"background: linear-gradient(transparent 60%,rgba(0,0,0,0.5) 0);\\" class=\\"vk_highlighter\\">1</span></p>
<!-- /wp:paragraph -->"
`;
